components/ws-manager-bridge/src/bridge.ts

```diff
@@ -371,11 +375,32 @@ export class WorkspaceManagerBridge implements Disposable {
                     .trace(ctx)
                     .findVolumeSnapshotById(status.conditions.volumeSnapshot.volumeSnapshotName);
                 if (existingSnapshot === undefined) {
-                    await this.workspaceDB.trace(ctx).storeVolumeSnapshot({
+                    const storedVolumeSnapshot = await this.workspaceDB.trace(ctx).storeVolumeSnapshot({
                         id: status.conditions.volumeSnapshot.volumeSnapshotName,
                         creationTime: new Date().toISOString(),
                         volumeHandle: status.conditions.volumeSnapshot.volumeSnapshotHandle,
                     });
+
+                    // since we stored most recent volume snapshot now, delete any old volume snapshots
```

Comment:
Uff. IMo the call _back_ to ws-manager-bridge for cleanup purposes is something we should _not_ handle here. :confused: It's already complicated and untestable as-is, but we should try hard to not make it worse.

Instead, I propose we have add a garbage-collection mechanism to `server`, in the same style that we use for workspaces, for instance. Optionally, this can be triggered by "workspace stopped" events. But has the additional benefit of being more reliable.

@sagor999 WDYT?

Reply:
@geropl feedback addressed!